### PR TITLE
fix strict countryCode problem

### LINKED ISSUE #674: strict countryCode problem
countryCode is the system region option which can be changed by the user, lots of users would set this region to where he lives. for example, in my system(macOS) language is en,  my region is chinese, then my deviceLocale is en_CN. i think countryCode is designed to be used for precisely specifying the language texts for each region. since user option of region can not be expected as in a range, it can be any option, but it's impossible to supply language texts for all regions (hundreds of countries). i think, it' better to fallback to the same language without country code first, then the global fallback option. 
For example, fallback locale is zh_CN(any other locale), when en_FR.json(a user who uses English and lives in France) is not found, it will fallback to zh_CN, but actually it's better to fallback to en.json(the same language without country code)

For a temporary solution, i use Locale('en') instead of Locale('en', 'US') and set useOnlyLangCode: true. 
- use Locale('en'), because the countryCode==null will make the locale.supports(anotherLocale) skips the countryCode check (i think it's better to skip this check by useOnlyLangCode == true)
- useOnlyLangCode: true to make it loading the en.json instead of en_REGION.json

But in this solution, it's not possible to specify the language texts for each region, for example, en_US, en_GB

### PR DESCRIPTION
**Pull request by PatrickChrestin**
Fix #674 

I added the functionality in that was described and discussed in the issue ticket.
I also added two tests based on the scenario mentioned in the issue ticket.

Feel free to comment or make suggestions.

@bw-flagship please have a look

### CODE REVIEW

File: lib/src/easy_localization_controller.dart [hunk 1]
```diff
@@ -207,9 +224,11 @@ class EasyLocalizationController extends ChangeNotifier {
   Locale? get savedLocale => _savedLocale;
 
   Future<void> resetLocale() async {
-    final locale = selectLocaleFrom(_supportedLocales!, deviceLocale, fallbackLocale: _fallbackLocale);
+    final locale = selectLocaleFrom(_supportedLocales!, deviceLocale,
```
Comment (bw-flagship) [inline lib/src/easy_localization_controller.dart]: unwanted changes here?

File: lib/src/easy_localization_controller.dart [hunk 2]
```diff
@@ -146,9 +161,11 @@ class EasyLocalizationController extends ChangeNotifier {
     final result = <String, dynamic>{};
     final loaderFutures = <Future<Map<String, dynamic>?>>[];
 
-    // need scriptCode, it might be better to use ignoreCountryCode as the variable name of useOnlyLangCode 
-    final Locale desiredLocale =
-        useOnlyLangCode ? Locale.fromSubtags(languageCode: locale.languageCode, scriptCode: locale.scriptCode) : locale;
+    // need scriptCode, it might be better to use ignoreCountryCode as the variable name of useOnlyLangCode
+    final Locale desiredLocale = useOnlyLangCode
```
Comment (bw-flagship) [inline lib/src/easy_localization_controller.dart]: unwanted changes here?

File: test/easy_localization_test.dart [hunk 3]
```diff
@@ -526,13 +558,13 @@ void main() {
...
-            expect(
-              Localization.instance.plural('test_empty_fallback_plurals', 2),
-              '',
-            );
-            expect(printLog, isEmpty);
-          }));
+        printLog = [];
+        expect(
+          Localization.instance.plural('test_empty_fallback_plurals', 2),
+          '',
```
Comment (bw-flagship) [inline test/easy_localization_test.dart]: unwanted changes here?

**Review by bw-flagship — CHANGES_REQUESTED** [review]
Thanks for the PR! I like the change and the tests, there are just some formatting changes that need to be reverted

**Comment by PatrickChrestin** [comment]
Perfect. I will take care of them when when I am back home from work.

**Comment by PatrickChrestin** [comment]
The changes to formatting should be all resolved now.

**Review by bw-flagship — APPROVED** [review]

**Pull request merged**